scripts/support/runner/jlink.py

```diff
@@ -51,10 +59,19 @@ def do_add_parser(cls, parser):
         parser.add_argument('--gdb-port', default=DEFAULT_JLINK_GDB_PORT,
                             help='pyocd gdb port, defaults to {}'.format(
                                 DEFAULT_JLINK_GDB_PORT))
+        parser.add_argument('--commander', default='JLinkExe',
+                            help='J-Link Commander, default is JLinkExe')
+        parser.add_argument('--erase', default=True, action='store_false',
```

Comment:
This isn't the default for other runners -- mind explaining why it's set here?

At least in the applications I work with, it's convenient not to mass erase, so that you can keep other flash areas around, e.g. those containing per-unit network credentials, etc.

It also makes flashing a bootloader, then the application, take more steps to enable mass-erase when flashing the bootloader, but then disabling it for the application.

Reply:
The bootloader use case is the reason I implemented this as an option. I set the default to True because the jlink documentation doesn't explicitly say if the `loadbin` command erases flash pages before programming, though the script output suggests that it does. So, I felt that mass erasing by default would be less error prone.

Reply:
> the jlink documentation doesn't explicitly say if the loadbin command erases flash pages before programming, though the script output suggests that it does.

The documentation is pretty vague indeed. I eventually found this information from Segger outside of the user manual:

```
How to perform downloading into flash via J-Link Commander:

Connect J-Link to the PC.
Connect target system to J-Link
Start J-Link Commander.
Type the following commands:
  J-Link> device <devicename> // For a list of known devices, please refer to here
  J-Link> r
  J-Link> h
  J-Link> loadbin <PathToBinFile>, <programmingaddress>
J-Link Commander executes the flash download and prints out the time statistics on success.
```

Source: https://www.segger.com/products/debug-probes/j-link/technology/flash-download/

Since this doesn't require an erase, I think it'd be better to have this default to False and have `action='store_true'` (which is a little more intuitive since the option's name is `--erase`).

Reply:
I reached out to the Segger folks for clarification. If they confirm that `loadbin` implicitly does a sector erase like we suspect, then I'm fine changing the mass-erase default to false.

Reply:
The Segger folks confirmed that `loadbin` does an implicit erase. See #5899